### PR TITLE
Integrate TektonHub

### LINKED ISSUE #121: Tekton catalog integration
Tekton catalog provides a catalog of reusable tasks and pipelines. It would be good to integration it when user creates a pipelines.

### PR DESCRIPTION
**Pull request by lstocchi**
it resolves #121

### CODE REVIEW

**Comment by lstocchi** [comment]
@jeffmaury this is how it works. Still some issue to be solved and code to be refactored (i am sure there is duplicate code to be removed somewhere) but if you can give it a look before tomorrow lunch so to know if there is something to change (visually) before the demo
tektonhub

File: src/test/java/com/redhat/devtools/intellij/tektoncd/utils/model/ConditionConfigurationModelTest.java [hunk 1]
```diff
@@ -23,6 +29,14 @@
 
     @Test
     public void checkEmptyConditionModel() throws IOException {
+        ResourceApi resApi = new ResourceApi();
```
Comment (jeffmaury) [inline src/test/java/com/redhat/devtools/intellij/tektoncd/utils/model/ConditionConfigurationModelTest.java:32]: Seems to me this is useless

File: src/main/java/com/redhat/devtools/intellij/tektoncd/WindowToolFactory.java [hunk 2]
```diff
@@ -48,6 +49,7 @@ public void createToolWindowContent(@NotNull Project project, @NotNull ToolWindo
             tree.putClientProperty(Constants.STRUCTURE_PROPERTY, structure);
             tree.setCellRenderer(new NodeRenderer());
             tree.addTreeWillExpandListener(new TreeExpansionListener());
+            HubModel.getInstance();
```
Comment (jeffmaury) [inline src/main/java/com/redhat/devtools/intellij/tektoncd/WindowToolFactory.java:52]: What is the rationale ? Is model is loaded asynchronously then `getInstance` should return a `CompletableFuture` and should called when required

**Review by jeffmaury — APPROVED** [review]
See my comments inline.
Other remarks:
- when you install a task and repeat install of the same workflow, then warning message is not show
- when a task is installed, I don't see the notification
- we don't support installation as cluster tasks but that may be postponed to another issue
- in terms of UI, if we can shrink a little but: I think the Install buttons are a little but too large and the area for displaying a task in the list also

**Comment by lstocchi** [comment]
> See my comments inline.
> Other remarks:
> 
> * when you install a task and repeat install of the same workflow, then warning message is not show
> * when a task is installed, I don't see the notification
> * we don't support installation as cluster tasks but that may be postponed to another issue
> * in terms of UI, if we can shrink a little but: I think the Install buttons are a little but too large and the area for displaying a task in the list also

I think I covered everything but the clusterTask installation. Please give it a look. Thanks!!

**Comment by jeffmaury** [comment]
Looks better but still:
- can't see the notification
- I think UI can still be shrinked a little bit

image

**Comment by lstocchi** [comment]
> Looks better but still:
> 
> * can't see the notification

What do you mean by notification? A popup window in the right bottom corner? Or you don't see the notification message in the event log panel?

**Comment by jeffmaury** [comment]
> 
> 
> > Looks better but still:
> > 
> > * can't see the notification
> 
> What do you mean by notification? A popup window in the right bottom corner? Or you don't see the notification message in the event log panel?

The popup window

**Comment by lstocchi** [comment]
@jeffmaury resized the column a little bit more. I don't think i can go further otherwise the warning icon is hidden.
Also added the popup notification. Give it a look, thanks!!

**Review by jeffmaury — APPROVED** [review]
Notification is now displayed

**Pull request merged**